test/amqp/high_level_test.rb

```diff
@@ -134,4 +134,450 @@ def test_it_can_resubscribe_on_reconnect
+    end
+  end
+
+  def test_queue_bind_with_exchange_object
+    msgs = Queue.new
+    client = AMQP::Client.new("amqp://#{TEST_AMQP_HOST}").start
+    begin
+      exchange = client.fanout("test.fanout.objbind")
+      queue = client.queue("test.fanout.objbind.queue")
+
+      # Test queue.bind with Exchange object (not just string)
+      queue.bind(exchange, "") # Pass Exchange object directly
+
+      queue.subscribe do |msg|
+        msgs.push msg
+      end
+
+      # Publish to exchange and verify message routing
+      exchange.publish("message via exchange object bind", "")
+
+      msg = msgs.pop
+      assert_equal "message via exchange object bind", msg.body
+      assert_equal "test.fanout.objbind", msg.exchange_name
+
+      # Test unbind with Exchange object
+      queue.unbind(exchange, "")
+
+      # Publish again - should not reach queue after unbind
+      exchange.publish("message after unbind", "")
+
+      sleep 0.01
+      assert_raises(ThreadError) do
+        msgs.pop(true)
+      end
+
+      # Clean up
+      exchange.delete
+      queue.delete
+    ensure
+      client.stop
+    end
+  end
+
+  def test_queue_bind_with_string_vs_exchange_object
+    client = AMQP::Client.new("amqp://#{TEST_AMQP_HOST}").start
+    begin
+      exchange = client.fanout("test.queue.bind.string.vs.obj")
+      queue1 = client.queue("test.queue.bind.string")
+      queue2 = client.queue("test.queue.bind.object")
+
+      # Test queue.bind with string name
+      queue1.bind("test.queue.bind.string.vs.obj", "")
+
+      # Test queue.bind with Exchange object
+      queue2.bind(exchange, "")
+
+      # Both should work the same way
+      assert true, "Queue can bind to both string names and Exchange objects"
+
+      # Test unbinding
+      queue1.unbind("test.queue.bind.string.vs.obj", "")
+      queue2.unbind(exchange, "")
+
+      # Clean up
+      exchange.delete
+      queue1.delete
+      queue2.delete
+    ensure
+      client.stop
+    end
+  end
+
+  def test_exchange_bind_to_queue_object
+    msgs = Queue.new
+    client = AMQP::Client.new("amqp://#{TEST_AMQP_HOST}").start
+    begin
+      exchange = client.fanout("test.exchange.bind.to.queue")
+      queue = client.queue("test.exchange.bind.queue.obj")
+
+      # Test exchange.bind with Queue object
+      exchange.bind(queue, "") # Bind exchange to queue directly
+
+      queue.subscribe do |msg|
+        msgs.push msg
+      end
+
+      # Publish to exchange and verify message routing
+      exchange.publish("message via queue object bind", "")
+
+      msg = msgs.pop
+      assert_equal "message via queue object bind", msg.body
+      assert_equal "test.exchange.bind.to.queue", msg.exchange_name
+
+      # Test unbind with Queue object
+      exchange.unbind(queue, "")
+
+      # Publish again - should not reach queue
+      exchange.publish("message after unbind", "")
+
+      sleep 0.01
+      assert_raises(ThreadError) do
+        msgs.pop(true)
+      end
```

Comment:
`ThreadError` seems like a not so nice error in this case, but I guess it's how it behaves now. :)

Reply:
Skimmed too fast, sorry.